### PR TITLE
Roben/qol api adjustments

### PR DESCRIPTION
**Pull request by ron-b**

### CODE REVIEW

File: src/GridParser.ts [hunk 1]
```diff
@@ -1,60 +1,60 @@
...
         const gridData: (string | number | boolean)[][] = initialDataGrid.gridData;
         if (!gridData) {
             throw new Error(gridNotFoundErr);
         }
 
         const rows: string[][] = [];
-        for (const rowData of gridData) {
+        if (initialDataGrid.promoteHeaders) {
+            const row: string[] = [];
+            for (const prop in gridData[0]) {
```
Comment (mabezen) [inline src/GridParser.ts:25]: Not sure I understand this part. if promote shouldn't we just skip the first row when building the cell data?

File: src/GridParser.ts [hunk 2]
```diff
@@ -1,60 +1,60 @@
...
 
         return rows;
     }
 
-    private validateGridHeader(data: Grid) {
-        const headerData: ColumnMetadata[] = data.header;
-        if (!headerData) {
-            throw new Error(headerNotFoundErr);
+    private generateColumnNames(initialDataGrid: Grid): string[] {  
+        if (!initialDataGrid.promoteHeaders) {
```
Comment (mabezen) [inline src/GridParser.ts:48]: Isn't it the opposite?  initialDataGrid.promoteHeaders -> take the first row as headers

File: src/types.ts [hunk 3]
```diff
@@ -23,27 +23,26 @@ export interface Metadata {
 }
 
 export interface TableData {
-    columnMetadata: ColumnMetadata[];
+    columnNames: string[];
```
Comment (mabezen) [inline src/types.ts:26]: What is the plan with the column metadata? we don't have any more typed columns it's all calculated by the value?

Comment (shanialbeck) [reply src/types.ts:26]: Yup. We're not receiving the column types so all cells are autodetected

**Pull request merged**